file.go

```diff
@@ -1103,24 +1103,41 @@ func (f *File) FlattenBatches() (*File, error) {
 
 	// Set batches to valid state
 	wg := sync.WaitGroup{}
+	doneCh := make(chan struct{}, 1)
+	errCh := make(chan error, 1)
 
 	wg.Add(len(batchesByHeader))
 	for _, b := range batchesByHeader {
 		go func(b Batcher) {
 			defer wg.Done()
-			_ = b.Create()
+			if err := b.Create(); err != nil {
+				errCh <- err
+			}
 		}(b)
 	}
 
 	wg.Add(len(IATBatchesByHeader))
 	for _, b := range IATBatchesByHeader {
 		go func(b IATBatch) {
 			defer wg.Done()
-			_ = b.Create()
+			if err := b.Create(); err != nil {
+				errCh <- err
+			}
 		}(b)
 	}
 
-	wg.Wait()
+	go func() {
+		wg.Wait()
+		close(doneCh)
+	}()
+
+	// Block til we're done creating batches or an error occurs
+	select {
+	case <-doneCh:
+		break
+	case err := <-errCh:
+		return nil, fmt.Errorf("creating batch: %w", err)
```

Comment:
If we were discarding errors before wouldn't this break the ABI (application behavior) of `FlattenBatches()`? Errors that were previously swallowed are now reported which could break calling code.  

I agree we should be checking these, but I can see why we skipped errors before. It was easier to "toss a bunch of batches into a file" and fix them after flattening.

Reply:
I almost want to suggest we add a `FlattenBatchesWith(opts *FlattenBatchesOpts)` and have `FlattenBatches()` supply `nil`. 

`FlattenBatchesOpts` can have an `ExposeErrors bool` and in the future stuff like `MaxBatches int` which has been requested before.

Reply:
ohhh i like that idea

Reply:
I'm going to implement what you suggested

Reply:
Sounds good. Can we restore the error discarding behavior and merge this? A performance improvement (without changing ABI) can go out as a bug fix.

FlattenBatchesWith would require a minor release.

Reply:
Sure thing, I'll create an issue to track

Reply:
If this is a pattern you consider helpful or extending, I'm a huge fan of https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis